src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -2337,4 +2462,166 @@ RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+
+    internalDstImgSizes->width = width;
+    internalDstImgSizes->height = height;
+    internalRoiTensorPtrSrc->xywhROI.xy.x = 0 ;
+    internalRoiTensorPtrSrc->xywhROI.xy.y = 0 ;
+    internalRoiTensorPtrSrc->xywhROI.roiHeight = 1080;
+    internalRoiTensorPtrSrc->xywhROI.roiWidth = 1920;
+
+    Rpp32u maskSize = FOG_MAX_HEIGHT * FOG_MAX_WIDTH;
+    Rpp32u maskSizeFloat = maskSize * sizeof(Rpp32f);
+
+    Rpp32f *d_fogAlphaMaskPtr, *d_fogIntensityMaskPtr;
+    d_fogAlphaMaskPtr =  reinterpret_cast<Rpp32f*>(internalRoiTensorPtrSrc + 4);
+    d_fogIntensityMaskPtr = reinterpret_cast<Rpp32f*>(d_fogAlphaMaskPtr + maskSize);
+
+    Rpp32f *d_dstfogAlphaMaskPtr, *d_dstfogIntensityMaskPtr;
+    d_dstfogAlphaMaskPtr = reinterpret_cast<Rpp32f*>(d_fogIntensityMaskPtr + maskSize);
+    d_dstfogIntensityMaskPtr = reinterpret_cast<Rpp32f*>(d_dstfogAlphaMaskPtr + (height * width));
+
+    CHECK_RETURN_STATUS(hipMemcpyAsync(d_fogAlphaMaskPtr, &fogAlphaMask_1920_1080[0], maskSizeFloat, hipMemcpyHostToDevice,rpp::deref(rppHandle).GetStream())); // Copying fog alpha mask from host to device asynchronously
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+
+    rppt_resize_gpu(d_fogAlphaMaskPtr, fogMaskSrcDescPtr, d_dstfogAlphaMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle); // Performing GPU resize operation on fog alpha mask
+    
+    CHECK_RETURN_STATUS(hipMemcpyAsync(d_fogIntensityMaskPtr, &fogIntensityMask_1920_1080[0], maskSizeFloat, hipMemcpyHostToDevice, rpp::deref(rppHandle).GetStream())); // Copying fog intensity mask from host to device asynchronously
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+
+    rppt_resize_gpu(d_fogIntensityMaskPtr, fogMaskSrcDescPtr, d_dstfogIntensityMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle); // Performing GPU resize operation on fog intensity mask
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+
+    std::random_device rd;  // Random number engine seed
+    std::mt19937 gen(rd()); // Seeding rd() to fast mersenne twister engine
+    Rpp32u *maskLocArrHostX, *maskLocArrHostY;
+    maskLocArrHostX = reinterpret_cast<Rpp32u*>(d_dstfogIntensityMaskPtr + (height * width));
+    maskLocArrHostY = reinterpret_cast<Rpp32u*>(maskLocArrHostX + srcDescPtr->n);
+    for(int i = 0; i < dstDescPtr->n; i++)
+    {
+        std::uniform_int_distribution<> distribX(0, width - roiTensorPtrSrcHost[i].xywhROI.roiWidth);
+        std::uniform_int_distribution<> distribY(0, height - roiTensorPtrSrcHost[i].xywhROI.roiHeight);
+        maskLocArrHostX[i] = distribX(gen);
+        maskLocArrHostY[i] = distribY(gen);
+    }
```

Comment:
As discussed, lets move this offset generation inside the hip executor, instead of doing it here